detectors/node/opentelemetry-resource-detector-azure/test/detectors/AzureFunctionsDetector.test.ts

```diff
@@ -35,7 +37,51 @@ describe('AzureFunctionsDetector', () => {
     process.env.FUNCTIONS_EXTENSION_VERSION = '~4';
     process.env.WEBSITE_MEMORY_LIMIT_MB = '1000';
 
-    const resource = azureFunctionsDetector.detect();
+    const resource = detectResourcesSync({
+      detectors: [azureFunctionsDetector, azureAppServiceDetector],
+    });
+    assert.ok(resource);
+    const attributes = resource.attributes;
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.FAAS_NAME],
+      'test-function'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.CLOUD_PROVIDER],
+      'azure'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.CLOUD_PLATFORM],
+      'azure_functions'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.CLOUD_REGION],
+      'test-region'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.FAAS_INSTANCE],
+      'test-instance-id'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.FAAS_MAX_MEMORY],
+      '1000'
+    );
+    assert.strictEqual(
+      attributes[SemanticResourceAttributes.FAAS_VERSION],
+      '~4'
+    );
+  });
+
+  it('should not detect azure app service values', () => {
+    process.env.WEBSITE_SITE_NAME = 'test-function';
+    process.env.REGION_NAME = 'test-region';
+    process.env.WEBSITE_INSTANCE_ID = 'test-instance-id';
+    process.env.FUNCTIONS_EXTENSION_VERSION = '~4';
+    process.env.WEBSITE_MEMORY_LIMIT_MB = '1000';
+
+    const resource = detectResourcesSync({
+      detectors: [azureFunctionsDetector, azureAppServiceDetector],
+    });
```

Comment:
Now the two test cases ('should test functions values' and 'should not detect azure app service values') are identical, so I'm not sure of the point.

Perhaps just have the one test case and add an assertion that one of the Azure App Service-specific attributes (e.g. `SemanticResourceAttributes.SERVICE_INSTANCE_ID`) is *not* defined?

Reply:
That makes sense. I've unified the two tests and now check for the absence of a few Azure App Service specific attributes.